sharding-orchestration/sharding-orchestration-core/src/main/java/io/shardingsphere/orchestration/internal/registry/config/listener/SchemaChangedListener.java

```diff
@@ -0,0 +1,138 @@
+        if (ChangedType.DELETED == event.getChangedType()) {
+            return createDeleteChangedEvent(shardingSchemaName);
+        }
+        return new IgnoredChangedEvent();
+    }
+    
+    private boolean isValidNodeChangedEvent(final String shardingSchemaName, final String nodeFullPath) {
+        return configurationNode.getDataSourcePath(shardingSchemaName).equals(nodeFullPath) || configurationNode.getRulePath(shardingSchemaName).equals(nodeFullPath);
+    }
+    
+    private ShardingOrchestrationEvent createUpdateChangedEvent(final String shardingSchemaName, final DataChangedEvent event) {
+        if (existedSchemaNames.contains(shardingSchemaName)) {
+            return createChangedEventForExistedSchema(event, shardingSchemaName);
+        }
+        return createChangedEventForNewSchema(shardingSchemaName);
+    }
+    
+    private ShardingOrchestrationEvent createChangedEventForExistedSchema(final DataChangedEvent event, final String shardingSchemaName) {
+        if (event.getKey().equals(configurationNode.getDataSourcePath(shardingSchemaName))) {
+            return createDataSourceChangedEvent(shardingSchemaName, event);
+        }
+        return createRuleChangedEvent(shardingSchemaName, event);
+    }
+    
+    private DataSourceChangedEvent createDataSourceChangedEvent(final String shardingSchemaName, final DataChangedEvent event) {
+        return new DataSourceChangedEvent(shardingSchemaName, ConfigurationYamlConverter.loadDataSourceConfigurations(event.getValue()));
+    }
+    
+    private ShardingOrchestrationEvent createRuleChangedEvent(final String shardingSchemaName, final DataChangedEvent event) {
+        return isShardingRule(event) ? createShardingRuleChangedEvent(shardingSchemaName, event.getValue()) : createMasterSlaveRuleChangedEvent(shardingSchemaName, event.getValue());
+    }
+    
+    private boolean isShardingRule(final DataChangedEvent event) {
+        return event.getValue().contains("tables:\n");
+    }
+    
+    private ShardingRuleChangedEvent createShardingRuleChangedEvent(final String shardingSchemaName, final String ruleValue) {
+        return new ShardingRuleChangedEvent(shardingSchemaName, ConfigurationYamlConverter.loadShardingRuleConfiguration(ruleValue));
+    }
+    
+    private MasterSlaveRuleChangedEvent createMasterSlaveRuleChangedEvent(final String shardingSchemaName, final String ruleValue) {
+        return new MasterSlaveRuleChangedEvent(shardingSchemaName, ConfigurationYamlConverter.loadMasterSlaveRuleConfiguration(ruleValue));
+    }
+    
+    private ShardingOrchestrationEvent createChangedEventForNewSchema(final String shardingSchemaName) {
+        if (isSufficientToInitialize(shardingSchemaName)) {
+            existedSchemaNames.add(shardingSchemaName);
+            return createSchemaChangedEvent(shardingSchemaName);
+        }
+        return new IgnoredChangedEvent();
+    }
+    
+    private boolean isSufficientToInitialize(final String shardingSchemaName) {
```

Comment:
What's mean of this method? maybe change a new name is better

Reply:
It is renamed to isOwnCompleteConfigurations, is't ok？